rclcpp/src/rclcpp/node_interfaces/node_base.cpp

```diff
@@ -310,3 +316,51 @@ NodeBase::resolve_topic_or_service_name(
   allocator.deallocate(output_cstr, allocator.state);
   return output;
 }
+
+rclcpp::node_interfaces::map_of_mutexes NodeBase::map_object = map_of_mutexes();
+
+rclcpp::node_interfaces::map_of_mutexes::map_of_mutexes()
+{
+}
+
+void rclcpp::node_interfaces::map_of_mutexes::create_mutex_of_nodebase(
+  const rclcpp::node_interfaces::NodeBaseInterface * nodebase)
+{
+  std::lock_guard<std::mutex> guard(this->internal_mutex);
+  this->data.emplace(nodebase, std::make_shared<std::mutex>() );
+}
+
+std::shared_ptr<std::mutex> rclcpp::node_interfaces::map_of_mutexes::get_mutex_of_nodebase(
+  const rclcpp::node_interfaces::NodeBaseInterface * nodebase)
+{
+  std::lock_guard<std::mutex> guard(this->internal_mutex);
+  return this->data[nodebase];
+}
+
+void rclcpp::node_interfaces::map_of_mutexes::delete_mutex_of_nodebase(
+  const rclcpp::node_interfaces::NodeBaseInterface * nodebase)
+{
+  std::lock_guard<std::mutex> guard(this->internal_mutex);
+  this->data.erase(nodebase);
+}
+
+rclcpp::node_interfaces::map_of_mutexes::~map_of_mutexes()
+{
+}
```

Comment:
Same here; just remove the default destructor.

Reply:
+1
I would also always stick to the rule of zero or rule of five, and never to the rule of three or any other intermediate solution.
See https://en.cppreference.com/w/cpp/language/rule_of_three#the_rule_of_three_five_zero#rule_of_zero, https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#c21-if-you-define-or-delete-any-copy-move-or-destructor-function-define-or-delete-them-all.